### PR TITLE
Add OCI standard labels to podman images

### PR DESCRIPTION
**Pull request by cevich**
Given a local container image, the OCI labels are very useful in tracking down the source and revision from whence it came.  Tooling like Renovate is also able to make use of these labels to suggest when newer versions are available.

Note: The current OCI spec. references defining these as annotations, however in practice, virtually nobody uses them.  Simple labels are more much accessible to both humans and tooling (like Renovate).

Update the podman container images README section to reflect the present-day reality.

### CODE REVIEW

**Comment by cevich** [comment]
[New labeling seems to have worked](https://cirrus-ci.com/task/4558723421569024), a skopeo inspect shows:

```json
  "Labels": {                                            
    "io.buildah.version": "1.33.5",
    "license": "MIT",                                    
    "name": "fedora",                                    
    "org.opencontainers.image.documentation": "https://github.com/containers/automation_images/blob/be4d07dbe566bb8a4600e3297ffe32478a5f95f2/README.md#container-images-overview-step-2",
    "org.opencontainers.image.revision": "be4d07dbe566bb8a4600e3297ffe32478a5f95f2",
    "org.opencontainers.image.source": "https://github.com/containers/automation_images",
    "org.opencontainers.image.url": "https://cirrus-ci.com/task/4558723421569024",
    "org.opencontainers.image.version": "20240410t155447z-f39f38d13",
    "vendor": "Fedora Project",
    "version": "39"                                      
  },
```

@edsantiago PTAL when you have a moment, this isn't critical.

File: podman/Containerfile [hunk 1]
```diff
@@ -15,6 +15,19 @@ ARG PACKER_BUILD_NAME=
...
+ARG IMG_SFX=
+ARG CIRRUS_TASK_ID=
+ARG GIT_HEAD=
+# Ref: https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL org.opencontainers.image.url="https://cirrus-ci.com/task/${CIRRUS_TASK_ID}"
+LABEL org.opencontainers.image.documentation="https://github.com/containers/automation_images/blob/${GIT_HEAD}/README.md#container-images-overview-step-2"
+# Renovate is sensitive to the following, they must point at the repo. and reflect commited
+# git sha's and current or future repo. tags/releases.
+# Ref: https://docs.renovatebot.com/modules/datasource/docker/
+LABEL org.opencontainers.image.source="https://github.com/containers/automation_images"
```
Comment (edsantiago) [inline podman/Containerfile:27]: There are eleven Containerfiles in this repo. Today. Could you add `/podman/Containerfile`?

File: README.md [hunk 2]
```diff
@@ -264,13 +264,11 @@ then automatically pushed to:
...
-* https://quay.io/repository/libpod/debian_podman
 
 The meaning of *prior* and *current*, is defined by the contents of
-the `*_release` files within the `podman` subdirectory.  This is
-necessary to support the Makefile target being used manually
-(e.g. debugging).  These files must be updated manually when introducing
-a new VM image version.
+the `*_RELEASE` values in the `Makefile`.  The images will be tagged
+with the value within the `IMG_SFX` file.  Additionally, the most
+recently merged PR on this repo. will tag it's images `latest`.
```
Comment (edsantiago) [inline README.md:271]: its, no apostrophe

File: podman/Containerfile [hunk 3]
```diff
@@ -15,6 +15,19 @@ ARG PACKER_BUILD_NAME=
 ENV AI_PATH=/usr/src/automation_images \
     CONTAINER=1
 
+ARG IMG_SFX=
+ARG CIRRUS_TASK_ID=
+ARG GIT_HEAD=
+# Ref: https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL org.opencontainers.image.url="https://cirrus-ci.com/task/${CIRRUS_TASK_ID}"
+LABEL org.opencontainers.image.documentation="https://github.com/containers/automation_images/blob/${GIT_HEAD}/README.md#container-images-overview-step-2"
+# Renovate is sensitive to the following, they must point at the repo. and reflect commited
```
Comment (edsantiago) [inline podman/Containerfile:24]: "repo" is a standard term, not an abbreviation. Would you mind removing the period? Every time I see repo-with-a-period I think it's the end of a sentence, and then the next sentence makes no sense.

**Review by edsantiago — COMMENTED** [review]
Thank you! Nonblocking requests inline.

File: podman/Containerfile (hunk 1 above)
Comment (cevich) [reply podman/Containerfile:27]: I originally had it this way, but was afraid Renovate will get mad.  Renovate isn't currently used for this image, so I can put it back.  Maybe it'll be okay?

File: README.md (hunk 2 above)
Comment (cevich) [reply README.md:271]: I'm cursed :rofl:

File: podman/Containerfile (hunk 3 above)
Comment (cevich) [reply podman/Containerfile:24]: I'll just remove this comment, it's not needed.

**Comment by cevich** [comment]
Force-push: Addressed all feedback items.  Pointed the source label at the exact commit's Containerfile to remove any doubt where/when it came from.

**Comment by edsantiago** [comment]
LGTM!

**Comment by edsantiago** [comment]
Github was briefly down, so I don't think we'll see the magic table. Merging anyway, thanks again.

**Comment by cevich** [comment]
Thanks for the suggestion!

**Pull request merged**